frontend/javascripts/oxalis/model/sagas/dataset_saga.ts

```diff
@@ -31,3 +32,29 @@ export function* watchMaximumRenderableLayers(): Saga<void> {
 
   yield* takeEvery(["WK_READY", "UPDATE_LAYER_SETTING", "UPDATE_DATASET_SETTING"], warnMaybe);
 }
+
+let downsamplingWarningDidShow = false;
+export function* watchZ1Downsampling(): Saga<void> {
+  function* maybeShowWarning(): Saga<void> {
+    const currentRes = yield* select((state) => getCurrentResolution(state));
+    const currentZoomStep = yield* select((state) => state.flycam.zoomStep);
+    const minVoxelPerPixel = 0.5;
+
+    if (!downsamplingWarningDidShow) {
+      // checking the downsampled dimensions x and y
+      for (let i = 0; i < 2; i++) {
+        const voxelPerPixelXY = currentZoomStep / currentRes[i];
+        if (voxelPerPixelXY < minVoxelPerPixel) {
+          Toast.warning(messages["dataset.z1_downsampling_hint"]);
```

Comment:
Additionally, I'd suggest to pass an `onClose` handler to `Toast.warning` like this:

```
Toast.warning(..., {
  key: ...
  onClose: () => {
    userClosedWarning = true
  }
}
```

`userClosedWarning` would replace `downsamplingWarningDidShow`. That way, the behavior should be as follows:

- the warning shows and hides automatically depending on how the current quality condition is (it can appear multiple times)
- if the user explicitly closed the notification via the X button, the onClose handler will be triggered which sets the boolean. afterwards, the toast won't show again due to the boolean.

I think, this behavior would be quite cool :)

Reply:
Agreed! Thanks for the idea :)